packages/wrangler/src/deployment-bundle/esbuild-plugins/hybrid-nodejs-compat.ts

```diff
@@ -193,16 +193,29 @@ function handleNodeJSGlobals(
  * This is important to support `inject` config for `performance` and `Performance` introduced
  * in https://github.com/unjs/unenv/pull/257
  */
-function encodeToLowerCase(str: string): string {
-	return str.replaceAll(/[A-Z]/g, (letter) => `$${letter.toLowerCase()}`);
+export function encodeToLowerCase(str: string): string {
+	return str
+		.replaceAll(/\$/g, () => "$$")
+		.replaceAll(/[A-Z]/g, (letter) => `$${letter.toLowerCase()}`);
 }
 
 /**
  * Decodes a string lowercased using `encodeToLowerCase` to the original strings
  */
-function decodeFromLowerCase(str: string): string {
-	return str.replaceAll(
-		/\$([a-z])/g,
-		(match, letter) => `${letter.toUpperCase()}`
-	);
+export function decodeFromLowerCase(str: string): string {
+	let out = "";
+	let i = 0;
+	while (i < str.length - 1) {
```

Comment:
since strings are iterable you can use the following to make the code simpler/more readable
```suggestion
	for (const letter of str) {
	}
```

Reply:
ah.. but then you can't use the i++ to advance and uppercase the next letter...

Reply:
nevermind... :)

Reply:
Right. I started doing exactly that.
And then I thought, well I could use a `continue` statement but then I have to track the state in the loop.
So I ended up falling back on the good old `i++` style for loop.